qubesbuilder/cli/cli_cleanup.py

```diff
@@ -79,15 +81,16 @@ def build_artifacts(obj: ContextObj, keep_versions):
             continue
 
         versions = sorted(
-            [v for v in component_dir.iterdir() if v.name != "noversion"],
-            reverse=True,
+            [v.name for v in component_dir.iterdir() if v.name != "noversion"],
+            reverse=True, key=Version
```

Comment:
`packaging.version.Version` is too strict by default:
```
>>> packaging.version.Version("4.3.1-1.50")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.12/site-packages/packaging/version.py", line 200, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
packaging.version.InvalidVersion: Invalid version: '4.3.1-1.50'
```

Reply:
Hum do we have this kind of version?

Reply:
Yes, it's a thing with `increment-devel-versions: true`, but also in qubes-release for example.

Reply:
Ah yes indeed, maybe we can change that with better pattern?

Reply:
for example: `4.3.1-1+devel50` or `4.3.1-1+50` are accepted by packaging:
```
>>> packaging.version.Version("4.3.1-1+devel50")
<Version('4.3.1.post1+devel50')>
>>> packaging.version.Version("4.3.1-1+50")
<Version('4.3.1.post1+50')>
```

Reply:
First, fractional release number is a valid thing, it is not only in devel versions.
But also, "post1" is not the same as "-1".

Anyway, it looks like we already have `QubesVersion` in component.py